main/timeseries.py

```diff
@@ -116,3 +123,44 @@ def get_capacity_timeseries(
         timeseries = update_timeseries(timeseries, capacity, "value")
 
     return timeseries
+
+
+def get_cost_recovery_timeseries(dates: list[tuple[date, date]]) -> pd.Series[float]:
+    """Get the cost recovery timeseries for the previous year.
+
+    For each month in the past year, this function aggregates all monthly charges and
+    divides this by the daily rate (dependent on funding source) and the number of
+    working days. This value is summed across all funding sources and added to the
+    timeseries.
+
+    Returns:
+        Pandas series containing cost recovery timeseries data.
+    """
+    date_range = pd.bdate_range(
+        start=dates[0][0],
+        end=dates[-1][1],
+        inclusive="both",
+    )
+    # initialize timeseries
+    timeseries = pd.Series(0.0, index=date_range)
+
+    for month in dates:
+        month_dates = pd.bdate_range(start=month[0], end=month[1], inclusive="both")
+        n_working_days = len(month_dates)
+        charges = (
+            models.MonthlyCharge.objects.filter(date=month[0])
+            .values("funding")  # group by funding
+            .annotate(total=Sum("amount"))  # get total Amount across monthly charges
+            .annotate(  # divide total by daily rate and working days
+                recovered=ExpressionWrapper(
+                    F("total") * Value(1.0) / F("funding__daily_rate") / n_working_days,
+                    output_field=FloatField(),
```

Comment:
Just to make sure I get this right: if there were just one person working, and all their time were invested in project work, then this `recovered` would be 1 for the month, right? And the maximum, if we had the whole team working full time in projects would be equal to the number of team members, correct?